### ISSUE TITLE
debug_traceTransaction on failed transaction

### ISSUE THREAD

**Issue by jsvisa** [celo-org/celo-blockchain#1942]
## Description

Failed to run `debug_traceTransaction` on failed transaction

## Current Behavior

I'm running a v1.6.10 archive node of Celo, but it seems there's something wrong on the tace model,

> send the request

```
http http://127.0.0.1:8545 jsonrpc=2.0 method=debug_traceTransaction params:='["0xdadb528f757df61b56322d326a38aa0c5e9db93f632d3c4c3e344bbf8bd03d7c", {"tracer": "callTracer", "timeout": "120s"}]' id=67
```
> got the response

```
{
    "jsonrpc": "2.0",
    "id": 67,
    "error": {
        "code": -32000,
        "message": "TypeError: cannot read property 'calls' of undefined    in server-side tracer function 'fault'"
    }
}
```

## Context

_If it makes sense, explain:_
 * _why is this important? Rationale for doing this_
 * _Is this part of a bigger initiatve, theme? Which?_

## Proposed Solution

BTW, maybe we need to cherry-pick the go-ethereum v1.10.xx's new  go-native tracer model.

**Comment by a26nine** [comment]
Facing the same issue with `debug_traceBlockByNumber` on `1.6.0`. Reverting to `1.5.8` resolves this.

![Screenshot 2022-10-06 at 20 39 16](https://user-images.githubusercontent.com/103454930/194414079-22b0657f-230e-4c3b-9a7d-e12ea80be3e3.png)

Another thing to note is that it didn't start at a specific block  — it’s distributed.

For e.g.,
15000000 / 0xe4e1c0 :x:
14950000 / 0xe41e70 :white_check_mark:
14900000 / 0xe35b20 :x:

**Comment by ymonye** [comment]
> Facing the same issue with `debug_traceBlockByNumber` on `1.6.0`. Reverting to `1.5.8` resolves this.
> 
> ![Screenshot 2022-10-06 at 20 39 16](https://user-images.githubusercontent.com/103454930/194414079-22b0657f-230e-4c3b-9a7d-e12ea80be3e3.png)
> 
> Another thing to note is that it didn't start at a specific block — it’s distributed.
> 
> For e.g., 15000000 / 0xe4e1c0 x 14950000 / 0xe41e70 white_check_mark 14900000 / 0xe35b20 x

Confirmed v1.5.8 resolves this.

**Comment by a26nine** [comment]
Looks like it's an upstream [issue](https://github.com/ethereum/go-ethereum/pull/23667). Let's wait for an update...

**Comment by palango** [comment]
@jsvisa @a26nine @ymonye : @piersy cherry-picked a potential fix for this in #1968.

It would be helpful if you could verify that the fix solves the problems you encountered. There's a docker image with the fixes available at http://us.gcr.io/celo-testnet/geth:d1d4d002419678ad56b2e511f1a37fc6b13fad62 , otherwise we can also provide binaries.

**Comment by jsvisa** [comment]
> @jsvisa @a26nine @ymonye : @piersy cherry-picked a potential fix for this in #1968.
> 
> It would be helpful if you could verify that the fix solves the problems you encountered. There's a docker image with the fixes available at http://us.gcr.io/celo-testnet/geth:d1d4d002419678ad56b2e511f1a37fc6b13fad62 , otherwise we can also provide binaries.

@palango seems this issue is fixed.

BTW, Is there any plan to backport the go-ethereum go-native trace into celo?  The js's consuming too much CPU to run the tracing.

**Comment by a26nine** [comment]
> @jsvisa @a26nine @ymonye : @piersy cherry-picked a potential fix for this in #1968.
> 
> It would be helpful if you could verify that the fix solves the problems you encountered. There's a docker image with the fixes available at http://us.gcr.io/celo-testnet/geth:d1d4d002419678ad56b2e511f1a37fc6b13fad62 , otherwise we can also provide binaries.

A binary to test would helpful. 🙏🏻

**Comment by palango** [comment]
> A binary to test would helpful. 🙏🏻

What system and OS do you need? You can also checkout the branch and [build it yourself](https://github.com/celo-org/celo-blockchain#building-the-source).


> BTW, Is there any plan to backport the go-ethereum go-native trace into celo? The js's consuming too much CPU to run the tracing. If you're interested, I can have a try to implement this feature

Let me bring it up internally.

**Comment by a26nine** [comment]
> > A binary to test would helpful. 🙏🏻
> 
> @a26nine What system and OS do you need? You can also checkout the branch and [build it yourself](https://github.com/celo-org/celo-blockchain#building-the-source).

Okay. I can confirm this (#1968) works as well. 👍🏻

**Comment by a26nine** [comment]
@palango Any ETA for the PR merge and release?

**Comment by piersy** [comment]
@a26nine The PR has been merged, I'm not sure when this will make it into a release

**Comment by gastonponti** [comment]
Fixed in the v1.6.1 (https://github.com/celo-org/celo-blockchain/releases/tag/v1.6.1)
